src/main/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaReader.java

```diff
@@ -77,10 +78,17 @@ private XmlMetaReader() {
                         new ResourcesScanner()).getResources(Pattern.compile(".*\\.xml"));
         final Set<String> allMetadataSources = new HashSet<>(standardModuleFileNames);
         for (String packageName : thirdPartyPackages) {
-            final Set<String> thirdPartyModuleFileNames =
-                    new Reflections(packageName, new ResourcesScanner())
-                            .getResources(Pattern.compile(".*checkstylemeta-.*\\.xml"));
-            allMetadataSources.addAll(thirdPartyModuleFileNames);
+            try {
+                final Set<String> thirdPartyModuleFileNames =
+                        new Reflections(packageName, new ResourcesScanner())
+                                .getResources(Pattern.compile(".*checkstylemeta-.*\\.xml"));
+                allMetadataSources.addAll(thirdPartyModuleFileNames);
+            }
+            catch (ReflectionsException ex) {
+                if (!"Scanner ResourcesScanner was not configured".equals(ex.getMessage())) {
+                    throw ex;
```

Comment:
Can we do:
```
If (){
// Comment why we swallow exception
}else {throw ex;}
```

Reply:
````
XmlMetaReader:88 Rule:EmptyIfStmt Priority:3 Avoid empty if statements.
````

Reply:
See new commit.

Reply:
I am not sure why but I like easy to read code more than follow this Rule.
Let's see reaction on next rewiever on this.
If I am alone, ok to merge.

Reply:
@strkkk,  @pbludov , please share your opinion on this

Reply:
Maybe this comment should be extracted to a method with Javadoc? Something like
```
if (propagateResourceScannerException(ex)) {
     throw ex;
}
```

Reply:
I like solution proposed above

Reply:
@strkkk , do you mean solution from @pbludov ?

Reply:
yes

Reply:
What is the best way to suppress this? Can it be suppressed by saying an if statement with a comment, do I suppress by method, or the full class? Let me know what the suppression is as I am not sure how to generate a tree for PMD.

Reply:
So which way am I changing the code to?

Reply:
I think this one https://github.com/checkstyle/checkstyle/pull/10786#discussion_r704071396 if @romani does not mind

Reply:
I am ok with Pavel proposal.

Reply:
@romani @strkkk @pbludov Done.